acceptance/tests/resource/exec/should_accept_large_output.rb

```diff
@@ -0,0 +1,33 @@
+test_name "tests that puppet correctly captures large and empty output."
+
+@testfilename = master.tmpfile('should_accept_large_output').split('/')[-1]
+
+agents.each do |agent|
+  testfile =
+    if /windows/ =~ agent[:platform]
+      "C:/#{@testfilename}"
+    else
+      "/tmp/#{@testfilename}"
+    end
```

Comment:
I think you can use `host.tmpfile` which IIRC will return a platform-specific path that you can use later in the test. Might also add a teardown for deleting the file.